src/dsp_tools/analyse_xml_data/construct_and_analyze_graph.py

```diff
@@ -188,73 +188,75 @@ def _remove_edges_to_stash(
     edges_to_remove: list[Edge],
     all_edges: list[Edge],
     remaining_nodes: set[int],
-) -> list[XMLLink | ResptrLink]:
+) -> None:
     """
     This function removes the edges from the graph in order to break a cycle.
-    It returns the information that enables us to identify the links in the real data.
 
     Args:
         graph: graph
         edges_to_remove: edges that should be removed
         all_edges: all edges in the original graph
         remaining_nodes: indices of the nodes in the graph
-
-    Returns:
-        the links that should be stashed
     """
-    source, target = edges_to_remove[0].source_rx_index, edges_to_remove[0].target_rx_index
-    links_to_stash = [x.link_object for x in edges_to_remove]
-    # if only one (source, target) is entered, it removes only one edge, not all
-    # therefore we need as many entries in the list as there are edges between the source and target to break the cycle
-    to_remove_list = [(x.source_rx_index, x.target_rx_index) for x in edges_to_remove]
-    phantom_links = []
-    for link_to_stash in links_to_stash:
+    # find the normal edges to remove
+    normal_edges_to_remove = [(x.source_rx_index, x.target_rx_index) for x in edges_to_remove]
+    # if only one (source, target) is removed, it removes only one edge, not all
+    # therefore we need as many entries in the list as there are edges between the source and the target
+
+    # find the phantom edges to remove
+    phantom_edges_to_remove = []
+    for link_to_stash in [x.link_object for x in edges_to_remove]:
         if isinstance(link_to_stash, XMLLink):
-            phantom_links.extend(_find_phantom_xml_edges(source, target, all_edges, link_to_stash, remaining_nodes))
-    to_remove_list.extend(phantom_links)
-    graph.remove_edges_from(to_remove_list)
-    return links_to_stash
+            new_phantoms = _find_phantom_xml_edges(
+                edges_to_remove[0].source_rx_index,
+                edges_to_remove[0].target_rx_index,
+                all_edges,
+                link_to_stash,
+                remaining_nodes,
+            )
+            phantom_edges_to_remove.extend(new_phantoms)
+
+    # remove the normal edges and the phantom edges
+    all_edges_to_remove = normal_edges_to_remove + phantom_edges_to_remove
+    graph.remove_edges_from(all_edges_to_remove)
 
 
 def _find_phantom_xml_edges(
-    source: int,
-    target: int,
-    edges: list[Edge],
-    xml_instance: XMLLink,
+    source_rx_index: int,
+    target_rx_index: int,
+    all_edges: list[Edge],
+    xml_link_to_stash: XMLLink,
```

Comment:
to me, the old variables were clearer and more concise. Only `xml_instance` I would rename to `xml_link`